adaptive/domain.py

```diff
@@ -0,0 +1,743 @@
+import abc
+import functools
+import itertools
+from collections import defaultdict
+
+import numpy as np
+import scipy.linalg
+import scipy.spatial
+from sortedcontainers import SortedList
+
+from adaptive.learner.triangulation import Triangulation, circumsphere, point_in_simplex
+
+__all__ = ["Domain", "Interval", "ConvexHull"]
+
+
+class Domain(metaclass=abc.ABCMeta):
```

Comment:
I think it would be useful to add a class docstring with a description of the important properties of this object.